game/engine.py

```
@@ -174,6 +174,99 @@ def from_dict(cls, data):
         game.valid_moves_cache = {}
         return game
 
+    @classmethod
+    def from_fen(cls, fen: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for classmethod `from_fen`

(ANN206)